File: src/Ginder.Domain/Entities/Player.cs
﻿using System.Collections.Generic;
using Ginder.Domain.ValueObjects;

namespace Ginder.Domain.Entities
{
    public class Player : Entity
    {
        public string Name { get; set; }
        public int Idade { get; set; }

        public virtual LocationValueObject Location { get; set; }

        public ICollection<PlayerGame> Games { get; set; }

        public Player()
        {
            Games = new List<PlayerGame>();            
        }
    }
}

File: Ginder.Android/app/src/main/java/br/unifor/dispositivos/ginder/data/model/Player.kt
package br.unifor.dispositivos.ginder.data.model

import java.time.OffsetDateTime
import java.util.*

public abstract class Entity(val Id: UUID) {
}

public class Player(Id: UUID, val Name: String, val Idade: Int, val Location: LocationValueObject?, val Games: List<PlayerGame>)
    : Entity(Id) {
}

public class LocationValueObject (val Latidude: Double, val Longitude: Double) {

}